Update Converters.java

lightistor opened the pull request:
Clean extra character in Cluster Node Line Entry for Google Cloud Platform Memory Store Redis when using the Jedis Connector. Sample entries causing ClusterNode information does not define host and port: 3765733728631672640db35fd2f04743c03119c6 10.180.0.33:11003@16379, master - 0 1708041426947 2 connected 5462-10922

<!--

Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:

-->

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

pivotal-cla commented:
@lightistor Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-redis&pullRequestId=2846)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-redis&pullRequestId=2846) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@lightistor Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-redis&pullRequestId=2846)!

lightistor commented:
@christophstrobl and @mp911de this is a fix to handle the format of the Cluster Node Line Entry for Google Cloud Platform Memory Store Redis that we are integrating our services with.
is it possible for me get assistance in merging this as a nightly build so that it is publicly available as opposed to us having have to host this fork internally?

mp911de commented:
Please provide a test case so that we document what isn't working right now for you and what should work. Also, `source..replace` is a syntax error introduced with this PR.

lightistor commented:
> Please provide a test case so that we document what isn't working right now for you and what should work. Also, `source..replace` is a syntax error introduced with this PR.

fixed the typo, added the unit test (toSetOfRedisClusterNodesShouldConvertNodesWithEmptyHumanReadableClusterNameInGcpMemoryStoreCorrectly, uses the entry format CLUSTER_NODE_ENTRY_WITH_EMPTY_HUMAN_READABLE_CLUSTER_NAME_IN_GCP_MEMORY_STORE)
and added myself to the authors in the header (per the checklist)

lightistor commented:
So, what is the possibility of it getting merged one day @mp911de ?

mp911de commented on src/main/java/org/springframework/data/redis/connection/convert/Converters.java:
```diff
@@ -569,8 +570,9 @@ enum ClusterNodesConverter implements Converter<String, RedisClusterNode> {
 		public RedisClusterNode convert(String source) {
-
-			String[] args = source.split(" ");
+			String filteredSource = source.replace(", ", " ");
```
That won't do the job especially when auxilliary fields are appended, then the `port` part is immediately followed by aux fields.

mp911de commented:
This PR isn't going anywhere really and the proposed fix is not addressing the underlying issue. As it would require a lot of effort from our side to fix and address shortcomings, we're instead closing this PR without a merge.
